packages/sanity/src/core/validation/validators/genericValidator.ts

```diff
@@ -94,6 +94,10 @@ export const genericValidators: Validators = {
 
   custom: async (fn, value, message, context) => {
     const slowTimer = setTimeout(() => {
+      // prevents this warning from appearing in the CLI validate command
+      const isNodeJs = typeof process !== 'undefined' && process?.release?.name === 'node'
+      if (isNodeJs) return
```

Comment:
Would love to figure out a better pattern for this that doesn't involve environment checking but rather has some sort of option built in.